web-bundle/src/main/java/com/graphhopper/replica/ReplicaFlagEncoderFactory.java

```diff
@@ -6,20 +6,26 @@
 import com.graphhopper.routing.util.VehicleEncodedValues;
 import com.graphhopper.util.PMap;
 
+import java.util.Set;
+
 public class ReplicaFlagEncoderFactory extends DefaultFlagEncoderFactory {
+    private final Set<String> vehicleNamesWithCustomSpeeds;
+
+    public ReplicaFlagEncoderFactory(Set<String> vehicleNamesWithCustomSpeeds) {
+        this.vehicleNamesWithCustomSpeeds = vehicleNamesWithCustomSpeeds;
+    }
 
     // "configuration" contains values from the flag encoder field in GraphHopper config (e.g.
     // graph.flag_encoders: car|turn_costs=true). we don't need to alter this field across different config files, so we
     // instead use this class to apply the necessary customizations to the default flag encoder
     @Override
     public FlagEncoder createFlagEncoder(final String name, PMap configuration) {
-        if (name.startsWith("car")) {
```

Comment:
This is more of a note for a follow-up, and shouldn't be addressed here, but reviewing this code made me realize that we'll need to make some modifications here and in `ReplicaVehicleTagParserFactory` if we want to support multi-profile definitions for modes other than `car`, which might be pursued in the future when we put work into improving bike/ped routing. Maybe we could sneak a `todo` comment here for visibility?

Reply:
[Also added a note about this in my doc [here](https://www.notion.so/replicahq/Customizing-and-Testing-Routers-c8058667962e49cba21894a84bca4066#8af1c91ec37d4d86ac15d12fdb1237d9)]

Reply:
good call, i'll add in a TODO

Reply:
actually on second thought, i don't think this logic was necessary in the first place. we have to do some custom stuff in `ReplicaFlagEncoderFactory` and `ReplicaCustomSpeedsCarTagParser` to accept the nonstandard vehicle names used with custom speeds, but this logic isn't necessary for just using custom models since the standard vehicle names are still used there (e.g. the `car_local` and `car_freeway` profiles still just use the `car` vehicle). so i removed the unnecessary logic from those classes.

i did add a TODO in ReplicaVehicleTagParserFactory about supporting custom speeds for vehicles other than cars, as the code does currently assume that custom speed files are used to customize car speeds in particular.

Reply:
Ahh cool, I think I understand - so basically, with the latest commit you pushed here, we can use custom _profiles_ with any "mode" we want, but custom speeds are only available for cars? That's great, I think it's more likely speeds will only be tweaked for cars, so this might be all we need to do for potential use cases in the near future.

Reply:
yep, exactly! agree that we probably won't need custom speeds for vehicles other than car any time soon